homeassistant/components/onkyo/__init__.py

```diff
@@ -1 +1,25 @@
 """The onkyo component."""
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+PLATFORMS = [Platform.MEDIA_PLAYER]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
It would be nice to move the set up of the connection to the device here, and set the object to `hass.data` or use [runtime_data from the config entry](https://developers.home-assistant.io/blog/2024/04/30/store-runtime-data-inside-config-entry), but I don't mind the runtime data to be for a follow up.

If the connection to the device fails, we can raise `ConfigEntryNotReady` which will tell the user in a user friendly way that something is wrong and will try again on a later moment.

Reply:
Not a direct answer but: the onkyo devices support asynchronous communication, which allows for near instant responses; no more polling. This refactor would be great to do when I implement that, maybe as the first follow up. I tried setting up the connection here, but that led to instability due to possible multiple connections. The underlying library isn't really built to support that use case well.

Reply:
> possible multiple connections

I think that my question around this is answered with the comment about autodiscovery of other devices

Reply:
I think what you can do in `__init__.py` is just create the connection to the device using `eiscp.eISCP(host)` and if that raises an exception like `OSError`, we raise ConfigEntryNotReady, so it will try again later. You can then store the result of `eiscp.eISCP(host)` in the [config entry runtime data](https://developers.home-assistant.io/blog/2024/04/30/store-runtime-data-inside-config-entry), and reuse that in `media_player.py`